File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.Arrays;

class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        int[] intArr = new int[n];
        for(int i = 0 ; i < n ; i++){
            int num1 = arr1[i];
            int num2 = arr2[i];
            
            intArr[i] = num1 | num2;
        }
        
        for(int i = 0 ; i < n ; i++){
            String s = Integer.toString(intArr[i], 2);
            while(s.length() != n){
                s = "0" + s;
            }
            s = s.replace('1', '#');
            s = s.replace('0', ' ');
            answer[i] = s;
        }
        return answer;
    }
}

File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    for a,b in zip(arr1, arr2):
        binary = str(bin(a|b)[2:])
        binary = binary.replace("1","#")
        binary = binary.replace("0"," ")
        binary = " "*(n-len(binary)) + binary
        answer.append(binary)
        
        
    return answer